### PR TITLE
Log losses

### PR DESCRIPTION
**Pull request by vict0rsch**
* `trainer.log_losses`
* `test_comet.py`
* add documentation in `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,3 +65,62 @@ loaders = Dict({
     val: { rn: loader, rf: loader, sn: loader, sf: loader}
 })
 ```
+
+### losses
+
+`trainer.losses` is a dictionary mapping to loss functions to optimize for the 3 main parts of the architecture: generator `G`, discriminators `D`, domain classifier `C`:
+
+```python
+trainer.losses = {
```
Comment (vict0rsch) [inline README.md]: @adrienju so you know, this needs to be updated as we work on losses

File: README.md [hunk 2]
```diff
@@ -64,3 +65,62 @@ loaders = Dict({
...
+    "D":{}, # discriminator losses from the generator and true data
+    "C": l1 | l2 | CE # classifier should predict the right 1-h vector [rf, rn, sf, sn]
+}
+```
+
+## Logging on comet
+
+Comet.ml will look for api keys in the following order: argument to the `Experiment(api_key=...)` call, `COMET_API_KEY` environment variable, `.comet.config` file in the current working directory, `.comet.config` in the current user's home directory.
+
+If your not managing several comet accounts at the same time, I recommend putting `.comet.config` in your home as such:
```
Comment (vict0rsch) [inline README.md]: @adrienju you should do this for the tests to pass

File: README.md [hunk 3]
```diff
@@ -64,3 +65,62 @@ loaders = Dict({
...
+
+### Parameters
+
+Set `train.log_level` in your configuration file to control the amount of logging on comet:
+
+* `0`: no logging on comet
+* `1`: only aggregated losses (representational loss, translation loss, total loss)
+* `2`: all losses (aggregated + task losses + auto-encoding losses)
+
+### Tests
```
Comment (vict0rsch) [inline README.md]: @adrienju FYI

File: omnigan/trainer.py [hunk 4]
```diff
@@ -46,14 +45,40 @@ def __init__(self, opts, comet=False, verbose=0):
...
+        }, "unknown model to log losses {}".format(model_to_update)
+
+        losses = self.logger.losses.copy()
+        if self.opts.train.log_level == 1:
+            # Only log aggregated losses: delete other keys in losses
+            for k in self.logger.losses:
+                if k not in {"representation", "generator", "translation"}:
+                    del losses[k]
+        # convert losses into a single-level dictionnary
+        losses = flatten_opts(losses)
```
Comment (vict0rsch) [inline omnigan/trainer.py]: @adrienju check this function's docstring to get what it does, it's quite simple

File: README.md (hunk 1 above)
Comment (adrienju) [reply README.md]: Oh right ! I forgot to add doc here thanks !

**Pull request merged**